### ISSUE TITLE
mine_function alias doesn't work in salt-ssh

### ISSUE THREAD

**Issue by Semyazz** [saltstack/salt#27808]
According to docs I defined something like this in my rosters file

```
minion:
    host: 192.168.0.5
    user: ubuntu
    sudo: True
    mine_functions:
      say_hello:
        mine_function: cmd.run 
        cmd: echo hello2
```

and when I try the following I got 
```
$ salt-ssh 'minion' mine.update
minion:
  True
$ salt-ssh 'minion' mine.get 'minion' say_hello
TypeError encountered executing say_hello: 'FunctionWrapper' object is not callable
```

But the pure `cmd.run` definition, without aliast works fine

```
minion:
    host: 192.168.0.5
    user: ubuntu
    sudo: True
    mine_functions:
      cmd.run: ['echo "hello2"']
```

Additionally, when I finally make that alias work, how can I make this thing work in `salt-ssh` in SLS files? I defined earlier `network.ip_addrs` and using `salt-ssh 'minion' mine.get 'minion' network_ip_addrs` it worked fine , but when I tried the same thing in `my mine-test.sls

```
{% set ips = salt['mine.get']('minion', 'network.ip_addrs') %}
```

but the only thing when I exec it is:

```
Rendering SLS 'base:mine-test' failed: Jinja error: 'master_job_cache'
Traceback...
```

Salt versions dump
```
$ salt-ssh --versions
Salt Version:
           Salt: 2015.8.0
 
Dependency Versions:
         Jinja2: 2.8
       M2Crypto: Not Installed
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.9 (default, Apr  2 2015, 15:33:21)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.2
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: Ubuntu 15.04 vivid
        machine: x86_64
        release: 3.19.0-21-generic
         system: Ubuntu 15.04 vivid
```

**Comment by jfindlay** [comment]
@Semyazz, thanks for the report.  This looks like a salt-ssh bug.

**Comment by rallytime** [comment]
@Semyazz Can you give the fix in #34835 a try?
